experimental/dds/tree/src/interface-prototype/Examples.ts

```diff
@@ -0,0 +1,141 @@
+import { v4 } from 'uuid';
+// This file uses these as opaque id types:
+// the user of these APIs should not know or care if they are short IDs or not, other than that they must be converted to StableId if stored for use outside of the shared tree it was acquired from.
+// In practice, these would most likely be implemented as ShortId numbers.
+import { Definition, TraitLabel } from '../Identifiers';
+import { Side } from '../Snapshot';
+import { StableId, TreeNodeData } from './Anchors';
+import {
+	anchorDataFromNodeId,
+	Command,
+	CommandContext,
+	CommandId,
+	CommandRegistry,
+	PrefetchFilter,
+	root,
+	SharedTree,
+	TreeNodeViewReadonly,
+} from './Checkout';
+import { Place, TreeNode } from './MutableAnchors';
+
+//////////////// Command examples //////////////
+
+// Inserts a node with the specified Definition and identifier at the specified Place, and return it.
+export const insertExample = {
+	id: 'f73f004c-3b3e-42fe-b7c9-2e5e8793ca45' as CommandId,
+	run: (
+		context: CommandContext,
+		{ definition: def, identifier: id }: { definition: StableId; identifier: StableId },
+		{ place }: { place: Place }
+	): TreeNode => {
+		const definition = context.loadDefinition(def);
+		const identifier = context.loadNodeId(id);
+		const range = place.insert({ definition, identifier, traits: {} });
+		return range.start.adjacentNode(Side.After);
+	},
+};
+
+function newNodeId(): StableId {
+	return v4() as StableId;
+}
+
+export const doubleInsertExample = {
+	id: '08bac27d-632f-48bb-834a-90af8d67ca60' as CommandId,
+	run: (context: CommandContext, _: {}, { parent }: { parent: TreeNode }): TreeNode => {
+		const a = context.runCommand(
+			insertExample,
+			{ definition: context.stabilize(bar), identifier: newNodeId() },
+			{ place: parent.childrenFromTrait(testTrait).start }
+		);
+
+		return context.runCommand(
+			insertExample,
+			{ definition: context.stabilize(baz), identifier: newNodeId() },
+			{ place: a.childrenFromTrait(testTrait).start }
+		);
+	},
+};
+
+const filterAll = () => true;
+
+const fetchAll: PrefetchFilter = {
+	value: filterAll,
+	children: filterAll,
+	traitChunks: filterAll,
+};
+
+const commands: CommandRegistry = [insertExample, doubleInsertExample];
+
+// Some dummy schema related data.
+const foo: Definition = 'Foo' as Definition;
+const bar: Definition = 'Bar' as Definition;
+const baz: Definition = 'Bar' as Definition;
+
+const testTrait: TraitLabel = 'testTrait' as TraitLabel;
+
+export async function exampleApp(tree: SharedTree): Promise<void> {
+	// Perform a full checkout of all data.
+	const checkout = await tree.checkout(commands, fetchAll);
+
+	// Example app policy: this app just watches the tree, and adds bar(baz) subtree under any new `foo` node.
+	checkout.on('viewChange', (before, after) => {
```

Comment:
Is this a common pattern, having a single method for registering change handlers? I was expecting either separate ".onViewChange = " registrations or for us to pass an interface with all the handlers, i.e. using property names instead of strings to specify the changes being handled.

Reply:
welcome to TypedEventEmitter. Its typesafe with statically typed string constants that look unsafe, but are actually strongly typed, but also compile anyway if you get them wrong, and fails to type check the call back signature. This is how Node does its javascript eventing, and fluid uses that with some only partially typesafe static typing.

Note that this event is not new in this prototype and is the same event currently used by experimental/tree, and using string names here is how other fluid code does it, and gets better static typing that using our enum (which is better at making sure you get the correct value.).